zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumPeerConfig.java

```diff
@@ -721,6 +721,10 @@ private void setupMyId() throws IOException {
             throw new IllegalArgumentException("serverid " + myIdString
                     + " is not a number");
         }
+        //myid must be in [1, 255]
+        if (serverId < 1 || serverId > 255) {
```

Comment:
There is a validateServerId logic in EphemeralType, which is going to check the serverId is within [1, 254] because of the conflict with TTL node, maybe we should have those logic in the same place.